spec/minimax_rb_spec.rb

```diff
@@ -1,145 +1,217 @@
+    end
 
-  it "goes for column win" do
-    board = ["X", nil, "O", "X", "O", nil, nil, nil, nil]
-    maximizing_marker = "X"
-    opponent = "O"
-    args = {
-      board: board,
-      curr_player_marker: maximizing_marker,
-      opponent_marker: opponent
-    }
-
-    expect(MinimaxRB.best_move(args)).to eq 6
-  end
+    it "returns false for a not full board" do
+      board = ["X", "O", "X", nil, "X", "O", "O", "X", "O"]
+      empty_marker = nil
 
-  it "goes for diagonal win" do
-    board = ["O", nil, "X", "O", "X", nil, nil, nil, nil]
-    maximizing_marker = "X"
-    opponent = "O"
-    args = {
-      board: board,
-      curr_player_marker: maximizing_marker,
-      opponent_marker: opponent
-    }
-    expect(MinimaxRB.best_move(args)).to eq 6
-  end
+      expect(MinimaxRB.board_full?(board, empty_marker)).to eq false
+    end
 
-  it "returns true for a draw" do
-    board = ["X", "O", "X", "O", "X", "O", "O", "X", "O"]
-    marker = "X"
-    opponent = "O"
-    empty_marker = nil
+    it "returns the rows of a board" do
+      board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
+      expected = [[0, 1, 2], [3, 4, 5], [6, 7, 8]]
 
-    expect(MinimaxRB.draw?(board, marker, opponent, empty_marker)).to eq true
-  end
+      expect(MinimaxRB.rows(board)).to eq expected
+    end
 
-  it "returns true for a win" do
-    board = ["X", "X", "X", "O", "O", nil, nil, nil, nil]
-    marker = "X"
-    opponent = "O"
+    it "returns the columns of a board" do
+      board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
+      expected = [[0, 3, 6], [1, 4, 7], [2, 5, 8]]
 
-    expect(MinimaxRB.win?(board, marker, opponent)).to eq true
-  end
-
-  it "returns true for a full board" do
-    board = ["X", "O", "X", "O", "X", "O", "O", "X", "O"]
-    empty_marker = nil
-
-    expect(MinimaxRB.board_full?(board, empty_marker)).to eq true
-  end
-
-  it "returns the rows of a board" do
-    board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
-    expected = [[0, 1, 2], [3, 4, 5], [6, 7, 8]]
-
-    expect(MinimaxRB.rows(board)).to eq expected
-  end
-
-  it "returns the columns of a board" do
-    board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
-    expected = [[0, 3, 6], [1, 4, 7], [2, 5, 8]]
-
-    expect(MinimaxRB.columns(board)).to eq expected
-  end
-
-  it "returns the diagonals of a board" do
-    board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
-    expected = [[0, 4, 8], [6, 4, 2]]
-
-    expect(MinimaxRB.diagonals(board)).to eq expected
-  end
-
-  it "with the highest score/move pair, returns the move" do
-    expect(MinimaxRB.max_move({5 => 1})).to eq 5
-  end
+      expect(MinimaxRB.columns(board)).to eq expected
+    end
 
-  it "with the highest score/move pair, returns the score" do
-    expect(MinimaxRB.max_score({5 => 1})).to eq 1
+    it "returns the diagonals of a board" do
+      board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
+      expected = [[0, 4, 8], [6, 4, 2]]
+
+      expect(MinimaxRB.diagonals(board)).to eq expected
+    end
+  end
+
+  context "game rules" do
```

Comment:
Grouping specs together into some logical hierarchy can be a great gut check to see if you're starting to specify a separate responsibility.  In this case, it's really getting into the rules of TicTacToe and away from the Minimax algorithm.  Maybe a somewhat different interface would allow Minimax to be run on anything (TicTacToe, Chess, etc...) if Minimax has a well of asking the game/board that called it to identify win/lose/draw conditions?